ui-react/src/components/topBar/ReportProblemModal.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useState } from "react";
+import { Button, Form, Modal, Row, Col, Stack } from "react-bootstrap";
+import { openExternal } from "../../eventsBridge";
+import { repository } from "../../../package.json";
+import os from "os";
+import LogsCanvas from "./LogsCanvas";
+import {isValidEmail} from "../../utils/validations"
+
+type ReportProblemForm = {
+  title?: string;
+  email?: string;
+  details?: string;
+};
+
+type ReportProblemModalProps = {
+  show: boolean;
+  onClose: () => {};
+};
+
+function ReportProblemModal({
+  show,
+  onClose,
+}: ReportProblemModalProps) {
+  const [form, setForm] = useState<ReportProblemForm>({
+    title: "",
+    email: "",
+    details: "",
+  });
+
+  const [errors, setErrors] = useState({});
+  const [showLogs, setShowLogs] = useState(false);
+
+  const setField = (field: string, value: string) => {
+    setForm({ ...form, [field]: value });
```

Comment:
This solution may cause race condition.
The proper solution would be using callback that will receive the current state and return the next state.

```suggestion
    setForm(prevForm=>({ ...prevForm, [field]: value }));
```
Another valid solution is `useReducer`